src/libLookup/Lookup.cpp

```diff
@@ -2057,12 +2057,36 @@ void Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
     // Store Tx Block to disk
     bytes serializedTxBlock;
     txBlock.Serialize(serializedTxBlock, 0);
+    uint64_t blockNum = txBlock.GetHeader().GetBlockNum();
 
-    if (!BlockStorage::GetBlockStorage().PutTxBlock(
-            txBlock.GetHeader().GetBlockNum(), serializedTxBlock)) {
+    if (!BlockStorage::GetBlockStorage().PutTxBlock(blockNum,
+                                                    serializedTxBlock)) {
       LOG_GENERAL(WARNING, "BlockStorage::PutTxBlock failed " << txBlock);
       return;
     }
+
+    // If txblk not from vacaous epoch and is rejoining as ds node
+    if ((blockNum + 1) % NUM_FINAL_BLOCK_PER_POW != 0 &&
+        (m_syncType == SyncType::DS_SYNC ||
+         m_syncType == SyncType::GUARD_DS_SYNC)) {
+      // Coinbase
+      uint128_t rewards = txBlock.GetHeader().GetRewards();
+      LOG_GENERAL(INFO, "Update coin base for finalblock with blockNum: "
+                            << blockNum << ", reward: " << rewards);
+      m_mediator.m_ds->SaveCoinbase(txBlock.GetB1(), txBlock.GetB2(),
+                                    CoinbaseReward::FINALBLOCK_REWARD,
+                                    blockNum + 1);
+      // Need if it join immediately before vacaous. And will be used in
+      // InitCoinbase in final blk consensus in vacaous epoch.
+      m_mediator.m_ds->m_totalTxnFees += rewards;
+
+      if (!BlockStorage::GetBlockStorage().PutEpochFin(blockNum)) {
+        LOG_GENERAL(WARNING, "BlockStorage::PutEpochFin failed "
+                                 << m_mediator.m_currentEpochNum);
```

Comment:
Should we use `blockNum + 1` instead of `m_currentEpochNum`?

Reply:
Agree. Fixed